beamer/chains.py

```diff
@@ -0,0 +1,82 @@
+from dataclasses import dataclass
+from typing import Any
+
+from beamer.typing import ChainId
+
+
+@dataclass(frozen=True)
+class ChainDescriptor:
+    id: ChainId
+    ecosystem_name: str
+    name: str
+    type2: bool = False
+    bedrock: bool = False
+    testnet: bool = False
+    local: bool = False
+
+
+DESCRIPTORS = (
+    # Ethereum
+    ChainDescriptor(id=ChainId(1), ecosystem_name="ethereum", name="mainnet", type2=True),
+    ChainDescriptor(
+        id=ChainId(5), ecosystem_name="ethereum", name="goerli", type2=True, testnet=True
+    ),
+    ChainDescriptor(
+        id=ChainId(1337), ecosystem_name="ethereum", name="local", type2=True, local=True
+    ),
+    # Optimism
+    ChainDescriptor(
+        id=ChainId(10), ecosystem_name="optimism", name="mainnet", type2=True, bedrock=True
+    ),
+    ChainDescriptor(
+        id=ChainId(420),
+        ecosystem_name="optimism",
+        name="goerli",
+        type2=True,
+        bedrock=True,
+        testnet=True,
+    ),
+    ChainDescriptor(
+        id=ChainId(901),
+        ecosystem_name="optimism",
+        name="local",
+        type2=True,
+        bedrock=True,
+        local=True,
+    ),
+    # Arbitrum
+    ChainDescriptor(id=ChainId(42161), ecosystem_name="arbitrum", name="mainnet"),
+    ChainDescriptor(id=ChainId(421613), ecosystem_name="arbitrum", name="goerli", testnet=True),
+    ChainDescriptor(id=ChainId(412346), ecosystem_name="arbitrum", name="local", local=True),
+    # Polygon ZkEVM
+    ChainDescriptor(id=ChainId(1101), ecosystem_name="polygon_zkevm", name="mainnet"),
+    ChainDescriptor(id=ChainId(1442), ecosystem_name="polygon_zkevm", name="goerli", testnet=True),
+    ChainDescriptor(id=ChainId(1001), ecosystem_name="polygon_zkevm", name="local", local=True),
+    # Base
+    ChainDescriptor(
+        id=ChainId(84531), ecosystem_name="base", name="goerli", testnet=True, bedrock=True
+    ),
+)
```

Comment:
Nitpick: I think it's unfortunate that black reformats these lines as I think they are not _too_ long and would be easier to read if black were to just ignore this entire block. This can be done like
```python
# fmt: off
_DESCRIPTORS = ...
# fmt: on
```
per https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#code-style

Reply:
it is long :D at least flake8 says also :D

Reply:
i am keeping formatting here